SwiftPackage/Sources/AssessmentModelUI/Views/Steps/LikertScaleQuestionView.swift

```diff
@@ -0,0 +1,157 @@
+    @State var width: CGFloat = 0
+    
+    var body: some View {
+        ZStack(alignment: .top) {
+            ScaleBar(fraction: $viewModel.fraction)
+                .offset(x: 0, y: 22)
+                .frame(width: max(0, width - 48))
+                
+            HStack(spacing: 0) {
+                ForEach(viewModel.dots) { dot in
+                    VStack(spacing: 0) {
+                        DotView(dot: dot)
+                        Text("\(dot.value)")
+                    }
+                    .onTapGesture {
+                        viewModel.value = dot.value
+                    }
+                }
+            }
+            .widthReader(width: $width)
+        }
+        .onAppear {
+            viewModel.initialize(questionState)
+        }
+    }
+    
+    struct ScaleBar : View {
+        @Binding var fraction: Double
+        var body: some View {
+            GeometryReader { geometry in
+                ZStack(alignment: .leading) {
+                    Rectangle()
+                        .innerShadow(.sliderBackground)
+                    Rectangle()
+                        .fill(Color.sageBlack)
+                        .frame(width: geometry.size.width * fraction)
+                }
+            }
+            .frame(height: 4)
+        }
+    }
+    
+    struct DotView : View {
+        @ObservedObject var dot: LikertScaleViewModel.Dot
+        var body: some View {
+            ZStack {
+                Circle()
+                    .innerShadow(.likertDotBackground)
+                Circle()
+                    .fill(Color.sageBlack)
+                    .opacity(dot.selected ? 1 : 0)
+            }
+            .frame(width: 24, height: 24)
+            .padding(.all, 12)
+        }
+    }
+}
+
+extension Shape {
+    func innerShadow(_ fillColor: Color) -> some View {
+        self
+            .fill(fillColor)
+            .overlay(
+                self
+                    .stroke(fillColor, lineWidth: 1)
+                    .shadow(color: .black.opacity(0.25), radius: 1, x: 0, y: 2)
+                    .clipShape(self)
+        )
+    }
+}
+
+fileprivate struct PreviewLikertScaleQuestionStepView : View {
+    let question: SimpleQuestionStep
+    var body: some View {
+        LikertScaleQuestionView(QuestionState(question,
+                                              answerResult: AnswerResultObject(identifier: question.identifier,value: .integer(2)),
+                                              skipStepText: Text("Skip question")))
+            .environmentObject(PagedNavigationViewModel(pageCount: 5, currentIndex: 2))
+            .environmentObject(AssessmentState(AssessmentObject(previewStep: question)))
+    }
+}
+
+struct LikertScaleQuestionView_Previews: PreviewProvider {
+    static var previews: some View {
+        PreviewLikertScaleQuestionStepView(question: likertExample1)
+    }
+}
+
+let likertExample1 = SimpleQuestionStepObject(
```

Comment:
So, FYI on our call with them last Friday, Aaron Katz at NU got all pedantic that by definition a Likert scale is balanced with a neutral/zero value in the center and equal-but-opposite choices on either side, e.g. a 5-point scale with "Strongly disagree/Somewhat disagree/Neither agree nor disagree/Somewhat agree/Strongly agree", and anything else is just a numeric scale. I don't personally see a problem with calling the code for displaying all such scales "Likert" and it would behave identically regardless of what the range of values is and what the options are called in the UI, but maybe the example should be something that is in fact technically a Likert scale? ¯\\_(ツ)\_\/¯

Reply:
*sigh*

Reply:
Lynn has been calling this a Likert scale for months. I'm assuming that she's calling it that b/c that's what it's called in the requirements doc (though I'd have to check). 

I'm not sure that I understand the issue. My example has 5 dots with the min = 1 and the max = 5. min/max are inclusive for this design. Is it that the scale should be -2 to 2? If he felt that strongly, then he should have said something *before* Lynn started working on these designs months ago.

Reply:
Lynn has been calling this a Likert scale for months. I'm assuming that she's calling it that b/c that's what it's called in the requirements doc (though I'd have to check).

I'm not sure that I understand the issue. My example has 5 dots with the min = 1 and the max = 5. min/max are inclusive for this design. Is it that the scale should be -2 to 2? If he felt that strongly, then he should have said something before Lynn started working on these designs months ago.